pt_os_web_portal/backend/helpers/finalise.py

```diff
@@ -164,3 +166,41 @@ def do_firmware_update():
         "touch /tmp/.com.pi-top.pi-topd.pt-poweroff.reboot-on-shutdown",
         timeout=10,
     )
+
+
+def get_non_ap_ip_addresses() -> List:
+    ips = list()
+    for iface in ["wlan0", "eth0", "ptusb0"]:
+        ip = get_internal_ip(iface)
+        if ip.replace("Internet Addresses Not Found", ""):
+            ips.append(ip)
+    logger.info(f"Device IPs: {ips}")
+    return ips
+
+
+def disable_ap_mode() -> None:
+    logger.info("Function disable_ap_mode()")
+    try:
+        run_command("/usr/bin/wifi-ap-sta disable", check=False, timeout=20)
+    except Exception as e:
+        logger.error(f"disable_ap_mode(): {e}")
+
+
+def on_same_network(request) -> Dict:
+    logger.info("Function on_same_network()")
+    on_same_network = False
+    pi_top_ip = ""
+    client_ip = request.remote_addr.split(":")[-1]
+    for ip in get_non_ap_ip_addresses():
+        if ip.split(".")[:-1] == client_ip.split(".")[:-1]:
```

Comment:
oh - ok, that's a scenario I hadn't considered, just pushed a change that includes it...

About `ip.split(".")[:-1] == client_ip.split(".")[:-1]` ... it's basically a 'rustic' way to check for the first 3 members of the IP's and say that the IPs are on the same network... this assumes that the netmask is `255.255.255.0` so it'll definitely not work on all networks...

Before this PR goes in, that function needs to be re-made. I think we can retrieve the netmask of a particular network interface and use it to create a smarter function to determine if 2 IPs are on the same network or not, probably using the `ipaddress` module (see [here](https://docs.python.org/3/library/ipaddress.html)).

The other method we discussed during the week to do this (sending a request from the frontend app to the IPs reported by the pi-top to determine if the devices are on the same network or not) won't work because the pi-top is internally routing all networks. This will cause that if the user is connected through AP and "pings" the `wlan0` IP, it'll get a response... Not sure if this behaviour is on purpose or not, maybe it got added with AP mode....